torch/_tensor_str.py

```diff
@@ -355,17 +357,22 @@ def _str_intern(self):
     if self.layout != torch.strided:
         suffixes.append('layout=' + str(self.layout))
 
-    if self.grad_fn is not None:
-        name = type(self.grad_fn).__name__
+    # Use inp here to get the original grad_fn and not the one generated by the forward grad
+    # unpacking.
```

Comment:
I'm guessing you implemented this function the other way first (i.e., have `self` be the dual tensor, and then at the end test if `self.tangent` is not None and then print the tangent in this case. Why did this not work? The `inp` use here seems like a clear indication that, semantically, we should be in the business of printing out information about the dual tensor as a whole, and not the primal.